app/models/budget.rb

```diff
@@ -52,4 +56,14 @@ def iati_status
       errors.add(:budget_type, I18n.t("activerecord.errors.models.budget.attributes.funding_type.source_fund.#{parent_activity.source_fund_code}"))
     end
   end
+
+  private def direct_budget_providing_org_must_be_beis
+    errors.add(:providing_organisation_name, "Providing organisation for direct funding must be BEIS!") unless providing_organisation_id == Organisation.service_owner.id
```

Comment:
Could you `service_owner?` on the association instead?

`unless providing_organisation.service_owner?`